src/test/java/UploadReceiptServletTest.java

```diff
@@ -286,6 +303,10 @@ public void doPostThrowsIfInvalidFile() throws IOException {
         BLOB_KEY, "image/png", new Date(), INVALID_FILENAME, IMAGE_SIZE_1MB, HASH, null);
     when(blobInfoFactory.loadBlobInfo(BLOB_KEY)).thenReturn(blobInfo);
 
+    long timestamp = clock.millis() - 1234;
```

Comment:
this little block seems to be repeated a lot. maybe put it in a utils helper function?

Reply:
also why are you subtracting 1234? is this to put the timestamp in the past? maybe put a comment somewhere saying why you do this. or rename the variable